ports/stm32f4/common-hal/busio/SPI.c

```diff
@@ -383,20 +383,20 @@ void common_hal_busio_spi_unlock(busio_spi_obj_t *self) {
 
 bool common_hal_busio_spi_write(busio_spi_obj_t *self,
         const uint8_t *data, size_t len) {
-    HAL_StatusTypeDef result = HAL_SPI_Transmit (&self->handle, (uint8_t *)data, (uint16_t)len, 2);
+    HAL_StatusTypeDef result = HAL_SPI_Transmit (&self->handle, (uint8_t *)data, (uint16_t)len, 500);
```

Comment:
Use `HAL_MAX_DELAY` here to disable the timeout altogether. https://github.com/adafruit/stm32f4/blob/89a356f0b44883d55c45a5d22940777a44a2b881/STM32F4xx_HAL_Driver/Src/stm32f4xx_hal_spi.c#L990 The SPI slave has no way of slowing down the transaction so we don't need to worry about timing out.